### PR TITLE
Fixes #11705: Remove the use of serial in rules

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/11705

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Awesome PR, just the modification on Technique that should not be here

File: rudder-core/src/main/scala/com/normation/cfclerk/domain/Technique.scala [hunk 1]
```diff
@@ -77,6 +77,42 @@ final case class AgentConfig(
   , bundlesequence : Seq[BundleName]
 )
 
+
+/**
+ * A type that tells if the Technique supports directive by directive
+ * generation or not.
+ */
+sealed trait TechniqueGenerationMode {
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/cfclerk/domain/Technique.scala]: That should be done in the PR that add the new generation mode

File: rudder-core/src/main/scala/com/normation/rudder/db/Doobie.scala [hunk 2]
```diff
@@ -56,6 +56,7 @@ import com.normation.rudder.services.reports.RunAndConfigInfo
 import org.slf4j.LoggerFactory
 import doobie.util.log.ExecFailure
 import doobie.util.log.ProcessingFailure
+import scala.language.implicitConversions
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/db/Doobie.scala]: this seems unused

**Comment by ncharles** [comment]
This is awesome :)

File: rudder-core/src/main/scala/com/normation/rudder/db/Doobie.scala (hunk 2 above)
Comment (fanf) [reply rudder-core/src/main/scala/com/normation/rudder/db/Doobie.scala]: it's necessary for using the keyword without warning with -Xlint compiler flag

**Comment by fanf** [comment]
I believe we can close that one since #1780 is there, andsuperseed that PR.

**Pull request merged**